pages/admin/lessons/[lessonId]/[pageName]/index.tsx

```diff
@@ -0,0 +1,174 @@
+      lesson {
+        id
+      }
+      order
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+  order: number
+}
+type Modules = Module[]
+
+type ModulesTabProps = {
+  filteredModules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const ModulesTab = ({
+  filteredModules,
+  lessonId,
+  refetch
+}: ModulesTabProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Omit<Module, 'order'>) => setSelectedIndex(item.id)
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={filteredModules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={filteredModules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const router = useRouter()
+  const { pageName, lessonId } = router.query
+
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
+
+  const lesson = useMemo(() => {
+    if (lessons) {
+      const lessonFromParam = lessons.find(e => e.id === toNumber(lessonId))
+
+      if (lessonFromParam)
+        return {
+          title: lessonFromParam.title,
+          id: lessonFromParam.id
+        }
+    }
+
+    return { title: '', id: -1 }
+  }, [lessons, lessonId])
+  const modules = useMemo(
+    () => sortBy(get(modulesData, 'modules'), 'order'),
+    [modulesData]
+  )
+
+  const filteredModules = useMemo(
+    () => modules.filter(module => module.lesson.id === lesson.id),
+    [lesson.id, modules]
+  )
+
+  const modulesTab = useMemo(
+    () => (
+      <ModulesTab
+        filteredModules={filteredModules}
+        lessonId={lesson.id}
+        refetch={refetch}
+      />
+    ),
+    [pageName, lessonId, filteredModules, lesson.id]
+  )
+
+  const renderPage = () => {
+    if (pageName === 'modules') return modulesTab
+
+    return (
+      <h1>
+        For now, you can only access <code>/modules</code> page
+      </h1>
+    )
+  }
+
+  return (
+    <AdminLayout data={data}>
+      <main className={styles.container}>
+        <header>
+          <h1>Lesson Info</h1>
+          <Breadcrumbs
+            lesson={lesson}
+            setLesson={lesson => {
+              router.push(`${MAIN_PATH}/${lesson.id}/${pageName}`)
+            }}
+            lessons={lessons}
+          />
+        </header>
+        <section>
+          <AdminLessonNav
+            tabs={[
+              {
+                tabName: 'modules',
+                urlPageName: 'modules'
+              }
+            ]}
+            render={navItem => {
```

Comment:
This is starting to look like a component, so let's try to pass in a component here instead. Generally we always try to render components instead of running functions.

```js

const SideNavComponent = ({navItem}) => {
  const isSelected = navItem.urlPageName === pageName
  const className =
    navStyles[
      isSelected
        ? 'lessons_tabsNav__nav__item'
        : 'lessons_tabsNav__nav__item--inactive'
    ]
  
  return (
    <Link href={`${MAIN_PATH}/${lesson.id}/${navItem.urlPageName}`}>
      <a className={`${className} nav-pills`}>
        {toUpper(navItem.tabName)}
      </a>
    </Link>
  )
  }
  
return (
      <AdminLayout data={data}>
      <main className={styles.container}>
        <header>...</header>
        <section>
          <AdminLessonNav
            tabs={[
              {
                tabName: 'modules',
                urlPageName: 'modules'
              }
            ]} 
            Component={SideNavComponent}
            />
            ...
)
  )
  ```
  
  And then to render this component, make the change in this file: https://github.com/garageScript/c0d3-app/blob/master/components/admin/lessons/AdminLessonNav.tsx#L15
  
```js
  const AdminLessonNav: React.FC<Props> = ({ tabs, Component }: Props) => {
  return (
    <div>
      <div className={styles.lessons_tabsNav}>
        <div className={styles.lessons__tabsNav__nav}>
          {tabs.map((tab, i) => (
            <Component key={i} navItem={tab}></Component>
          ))}
        </div>
      </div>
    </div>
  )
}
```